ratelimit/ratelimit.go

```diff
@@ -336,9 +336,9 @@ func (s Settings) String() string {
 
 // limiter defines the requirement to be used as a ratelimit implmentation.
 type limiter interface {
-	// Allow is used to get a decision if you should allow the
-	// call to pass or to ratelimit
-	Allow(string) bool
+	// AllowContext is used to get a decision if you should allow the
+	// call with context to pass or to ratelimit
+	AllowContext(context.Context, string) bool
```

Comment:
Do we need "Context" in the name? 
Maybe it could be just `Allow(context.Context, string) bool`, wdyt?

@szuecs ?

Reply:
also I think we do not need "with context" addition in the docs - method obviously requires context

Reply:
Yes let's go with `Allow(context.Context, string) bool`